Native profiler v0.9.3 - allocation profiling (#14963)
DataDog/documentation: content/en/profiler/search_profiles.md

Four tabs are below the profile header:

| Tab               | Definition                                                                                                                            |
| ----------------- | ------------------------------------------------------------------------------------------------------------------------------------- |

> | Profiles          | A flame graph and summary table of the profile you are looking at. You can switch between profile types (`CPU`, `Memory allocation`). |

Reviewer: ```suggestion
| Profiles          | A flame graph and summary table of the profile you are looking at. You can switch between profile types (for example, `CPU`, `Memory allocation`). |
```

> | Analysis          | A set of heuristics that suggest potential issues or areas of improvement in your code. Currently only available for Java.            |

Reviewer: ```suggestion
| Analysis          | A set of heuristics that suggest potential issues or areas of improvement in your code. Only available for Java.            |
```

Block
: The time functions have been waiting on mutexes and channel operations during the profiling period (default: 60s). Sleep, GC, Network, and Syscall operations are not captured by this profile. Blocking operations are only captured after they become unblocked, so this profile cannot be used to debug applications that appear to be stuck. For mutex contentions, the stack traces in this profile point to blocked `Lock()` operations. This tells you where your program is getting blocked, while the mutex profile tells you what part of your program is causing the contention. See our [Block Profiling in Go][1] research for more in-depth information. See also the note about how this measure changes in version `1.33.0` in [Delta profiles](#delta-profiles). **Note:** The block profiler can cause noticeable overhead for production workloads. If enabling it in production, prefer high rates (such as `100000000`, which is 100 milliseconds) and look for signs of increased latency or CPU utilization.

Goroutines

> : A snapshot of the number of goroutines currently executing the same functions (both on-CPU and waiting off-CPU). An increasing number of goroutines between snapshots can indicate that the program is leaking goroutines. In most healthy applications this profile is dominated by worker pools and the number of goroutines they use. Applications that are extremely latency-sensitive and use a large number of goroutines (> 10.000) should be aware that enabling this profile requires stop-the-world pauses. The pauses occur only once every profiling period (default 60s) and normally last for `~1µsec` per goroutine. Typical applications with a p99 latency SLO of `~100ms` can generally ignore this warning. See our [Goroutine Profiling in Go][2] research for more in-depth information.

Reviewer: ```suggestion
: A snapshot of the number of goroutines currently executing the same functions (both on-CPU and waiting off-CPU). An increasing number of goroutines between snapshots can indicate that the program is leaking goroutines. In most healthy applications this profile is dominated by worker pools and the number of goroutines they use. Applications that are extremely latency-sensitive and use a large number of goroutines (> 10.000) should be aware that enabling this profile requires stop-the-world pauses. The pauses occur only once every profiling period (default 60s) and normally last for around `1µsec` per goroutine. Typical applications with a p99 latency SLO of around `100ms` can generally ignore this warning. See our [Goroutine Profiling in Go][2] research for more in-depth information.
```

CPU
: Shows the time each function spent running on the CPU.

Allocations

> : Shows the number of allocations by each method, including allocations which were subsequently freed.

Reviewer: ```suggestion
: The number of allocations by each method, including allocations which were subsequently freed.
```

> : Shows the number of allocations by each method, including allocations which were subsequently freed.

Reviewer 2: ```suggestion
: Shows the number of allocations done by each function in heap memory during the profiling period (default: 60s), including allocations which were subsequently freed. Stack allocations are not tracked.
```

Allocated memory

> : Shows the amount of heap memory allocated by each method, including allocations which were subsequently freed.

Reviewer: ```suggestion
: The amount of heap memory allocated by each method, including allocations which were subsequently freed.
```

Reviewer 2: ```suggestion
: Shows the amount of heap memory allocated by each function during the profiling period (default: 60s), including allocations which were subsequently freed. Stack allocations are not tracked. 
```
